src/main/java/slimeknights/tconstruct/tools/modifiers/ability/BlockTransformModifier.java

```diff
@@ -0,0 +1,46 @@
+package slimeknights.tconstruct.tools.modifiers.ability;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemUseContext;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.SoundEvent;
+import net.minecraftforge.common.ToolType;
+import slimeknights.tconstruct.library.modifiers.SingleUseModifier;
+import slimeknights.tconstruct.library.tools.item.ToolCore;
+import slimeknights.tconstruct.library.tools.nbt.ToolStack;
+
+public class BlockTransformModifier extends SingleUseModifier {
+  private final ToolType toolType;
+  private final SoundEvent sound;
+  private final boolean requireGround;
+  private final int priority;
+
+  public BlockTransformModifier(int color, int priority, ToolType toolType, SoundEvent sound, boolean requireGround) {
+    super(color);
+    this.priority = priority;
+    this.toolType = toolType;
+    this.sound = sound;
+    this.requireGround = requireGround;
+  }
+
+  @Override
+  public int getPriority() {
+    return priority;
+  }
+
+  @Override
+  public boolean shouldDisplay(boolean advanced) {
+    return priority > Short.MIN_VALUE;
+  }
+
+  @Override
+  public ActionResultType onItemUse(ToolStack tool, int level, ItemStack stack, ItemUseContext context) {
+    Item item = stack.getItem();
+    if (item instanceof ToolCore) {
+      ToolCore toolCore = (ToolCore) item;
+      return toolCore.getToolHarvestLogic().transformBlocks(context, toolType, sound, requireGround);

```

Comment:
I half wonder if this logic should move into the modifier instead of tool harvest logic. None of the tools should be overriding it, it was just in ToolHarvestLogic as a clean place to reuse it.

Reply:
...Maybe someone might want to AoE transform blocks in a different place, i dunno.

Reply:
Could always make it static here. I just don't see a ton of potential for reuse of that outside this modifier. We could always do it after this PR as part of library cleanup, its just not something strictly related to AOE.

Reply:
Either way, this is kind of out of scope of this PR anyway and it can always be discussed later. ToolHarvestLogic just feels like a nicer place for it.